### PR TITLE
test: schedule_api : test some kernel api

### PR DESCRIPTION
**Pull request by yingmingx**
1.In order to improve function coverage, add testcase
for kernel api api z_pend_thread() z_unpend_thread()
z_unpend_all() z_pend_curr_irqlock()
z_move_thread_to_end_of_prio_q() z_remove_thread_from_ready_q()

2.Add error test of api in sched.c to improve
branch and line coverage.

Signed-off-by: Ying ming <mingx.ying@intel.com>

### CODE REVIEW

File: tests/kernel/sched/schedule_api/src/test_sched_pending.c [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+{
+	z_waitq_init(&pending.waitq);
+
+	unsigned int key = irq_lock();
+	z_pend_curr_irqlock(key, &pending.waitq, K_FOREVER);
+}
+
+static void thread_handler3(void *p1, void *p2, void *p3)
+{
+	num++;
```
Comment (LeiW000) [inline tests/kernel/sched/schedule_api/src/test_sched_pending.c:94]: I'm wondering if we should add a kind of lock here?

**Review by LeiW000 — CHANGES_REQUESTED** [review]

File: tests/kernel/sched/schedule_api/src/test_sched_pending.c [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+{
+	k_thread_abort(tid);
+	
+	ztest_test_pass();
+}
+
+#ifdef CONFIG_ASSERT_NO_FILE_INFO
+void assert_post_action(void)
+#else
+void assert_post_action(const char *file, unsigned int line)
```
Comment (enjiamai) [inline tests/kernel/sched/schedule_api/src/test_sched_pending.c:42]: The common fatal and assert handler has been merged, could you please leverage them instead? thanks!  
please refer to https://github.com/zephyrproject-rtos/zephyr/pull/30150

**Review by enjiamai — CHANGES_REQUESTED** [review]
The common fatal and assert handler has been merged, could you please leverage them instead? thanks!  
please refer to https://github.com/zephyrproject-rtos/zephyr/pull/30150

File: tests/kernel/sched/schedule_api/src/test_sched_pending.c (hunk 1 above)
Comment (yingmingx) [reply tests/kernel/sched/schedule_api/src/test_sched_pending.c]: This is really a problem that needs attention. I have changed code and used atomic variable here.

**Comment by yingmingx** [comment]
> The common fatal and assert handler has been merged, could you please leverage them instead? thanks!
> please refer to #30150
Hi, enjia. All the assert related code have been removed and fatal framework have been used. Please help to give another look, thanks.

File: tests/kernel/sched/schedule_api/src/test_sched_pending.c [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+{
+	atomic_inc(&num);
+}
+
+/**
+ * @brief Test kernel api z_pend_thread()  z_unpend_thread()
+ * z_unpend_all() z_pend_curr_irqlock()
+ *
+ * @ingroup kernel_sched_tests
+ *
```
Comment (enjiamai) [inline tests/kernel/sched/schedule_api/src/test_sched_pending.c:56]: I know you have already commented the code. My suggestion is to add @details to descript the high level testing purpose of this case, thanks!

File: tests/kernel/sched/schedule_api/src/test_sched_pending.c [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+				K_PRIO_PREEMPT(0), 0, K_NO_WAIT);
+	k_msleep(1);
+	zassert_true(z_is_thread_pending(&tdata), NULL);
+	k_thread_abort(tid);
+}
+
+/**
+ * @brief Test kernel api z_move_thread_to_end_of_prio_q()
+ * z_remove_thread_from_ready_q()
+ *
```
Comment (enjiamai) [inline tests/kernel/sched/schedule_api/src/test_sched_pending.c:95]: Same as line 56.

**Review by enjiamai — CHANGES_REQUESTED** [review]
could you please add @details for your test cases, thanks!

File: tests/kernel/sched/schedule_api/src/test_sched_api_error.c [hunk 5]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
```
Comment (LeiW000) [inline tests/kernel/sched/schedule_api/src/test_sched_api_error.c:2]: Let change it with "2021".

**Review by LeiW000 — CHANGES_REQUESTED** [review]

**Review by LeiW000 — APPROVED** [review]

File: tests/kernel/sched/schedule_api/src/test_sched_pending.c (hunk 4 above)
Comment (yingmingx) [reply tests/kernel/sched/schedule_api/src/test_sched_pending.c:102]: Hi ,enjia. Details tag has been added and all the build and run error solved.

**Comment by yingmingx** [comment]
Close this PR and replace it with #32601

**Pull request closed without merging**